src/main/java/org/reaktivity/specification/socks/internal/Functions.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.specification.socks.internal;
+
+import java.nio.charset.StandardCharsets;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.kaazing.k3po.lang.el.Function;
+import org.kaazing.k3po.lang.el.spi.FunctionMapperSpi;
+import org.reaktivity.specification.socks.internal.types.String16FW;
+import org.reaktivity.specification.socks.internal.types.stream.TcpBeginExFW;
+
+public final class Functions
+{
+
+    @Function
+    public static byte[] destAddrPort(String destAddrPort)
+    {
+        MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[1024]);
+        String16FW dapType = new String16FW.Builder()
+            .wrap(writeBuffer, 0, writeBuffer.capacity())
+            .set(destAddrPort, StandardCharsets.UTF_8)
+            .build();
+        byte[] dapBytes = new byte[dapType.sizeof()];
+        dapType.buffer().getBytes(0, dapBytes);
+        return dapBytes;
+    }
+
+    /**
+     * Builds the TcpBeginExFW from given IPv4 addresses
+     * FIXME Should be implemented in tcp.spec nukleus, with tests on both client and server.
```

Comment:
Yes, this is needed so that the socks server would send back to the client the address and port it has used to connect to the requested address.

> CONNECT
> 
>    In the reply to a CONNECT, BND.PORT contains the port number that the
>    server assigned to connect to the target host, while BND.ADDR
>    contains the associated IP address.  The supplied BND.ADDR is often
>    different from the IP address that the client uses to reach the SOCKS
>    server, since such servers are often multi-homed.  It is expected
>    that the SOCKS server will use DST.ADDR and DST.PORT, and the
>    client-side source address and port in evaluating the CONNECT
>    request.